### ISSUE TITLE
Mask is not propagated into Sequential keras layers

### ISSUE THREAD

**Issue by mpdn** [tensorflow/tensorflow#31638]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac OS
- TensorFlow installed from (source or binary): docker, pip
- TensorFlow version (use command below): 1.14.0, 2.0.0b
- Python version: 3.6.8

**Describe the current behavior**

When a masked input is fed to a `tf.keras.models.Sequential`, the mask is ignored

**Describe the expected behavior**

I would expect the mask to be propagated as if the inner layers were not in a `Sequential` layer.

**Code to reproduce the issue**

This works:
```
import tensorflow as tf

def check_mask(inputs, mask=None):
    assert mask is not None
    return inputs

model = tf.keras.models.Sequential([
    tf.keras.layers.Masking(0, input_shape=[1]),
    tf.keras.layers.Lambda(check_mask)
])
```

This fails due to failed assert:
```
import tensorflow as tf

def check_mask(inputs, mask=None):
    assert mask is not None
    return inputs

model = tf.keras.models.Sequential([
    tf.keras.layers.Masking(0, input_shape=[1]),
    tf.keras.models.Sequential([
        tf.keras.layers.Lambda(check_mask),
    ]),
])
```

**Other info / logs**
I think I have narrowed this down to a cache invalidation issue in `tf.keras.layers.Layer._should_compute_mask`: https://github.com/tensorflow/tensorflow/blob/e19c354920c3b246dda6598229210a582caaa1a9/tensorflow/python/keras/engine/base_layer.py#L2055-L2059

From my debugging, it seems that whenever the issue above appears `_should_compute_mask` is cached as false even though evaluating the actual property expression results in true.

Since this problem is in a base layer, this issue might be affecting other layers as well.

**Comment by oanush** [comment]
I was able to replicate the issue for given code in both TF versions- 1.14 and 2.0beta. Thanks

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with [TF v2.1](https://colab.research.google.com/gist/amahendrakar/19fd10326257085b09f137049681b6be/2-1-template.ipynb) and [TF-nightly](https://colab.research.google.com/gist/amahendrakar/bbd1ef55fd48d1851f0b984a5513429c/tf-nightly.ipynb#scrollTo=ieAW-NK5iqpf) i.e. v2.2.0-dev20200327. Please find the attached gist. Thanks!

**Comment by saikumarchalla** [comment]
@mpdn Closing this issue as it is fixed in TF 2.x. Thanks!
